docs/cql/cql-extensions.md

```diff
@@ -232,6 +232,25 @@ to explicitly mark them as such.
 
 ## Expressions
 
+### NULL
+
+Scylla aims for a uniform handling of NULL values in expressions, inspired
+by SQL: The overarching principle is that a NULL signifies an absense of
+value, so any expression calculated based on a NULL also results in a NULL.
+For example, the result of `x + NULL`, `x = NULL`, `x < NULL` are all NULL,
+no matter what `x` is. Even the expression `NULL = NULL` returns NULL, not
+TRUE.
+
+From the last example we can see that the `SELECT` filter `WHERE x = NULL`
+matches (returns `TRUE`) no row. It doesn't even match rows where x is
+missing. To match rows with missing x, SQL offers the syntax `x IS NULL`
+(and similarly, also `x IS NOT NULL`). Scylla does not yet implement this
+syntax.
+
+In contrast, Cassandra is less consistent in its handling of nulls.
+The example `x = NULL` is considered an error, not a valid expression
+whose result is NULL.
```

Comment:
In LWT `IF` clauses, `x = NULL` is supported (and works like SQL `x IS NULL`).

Reply:
@cvybhu just pointed this out to me in another comment. This is terrible. We argued for a long time what "x = NULL" should mean, if you remember I was in the camp that "x = NULL should be useful, so it should check equality not do nothing" but the decision was to go for **consistency** and that `x=NULL` should match nothing. So how did we end up with this inconsistency after making the decision to go for consistency?

I see it's good I'm documenting this. If I got confused, there is no way users will remember this without a document.